corsheaders/middleware.py

```diff
@@ -57,9 +57,22 @@ def _https_referer_replace(self, request):
             if not conf.CORS_ORIGIN_ALLOW_ALL and not self.origin_found_in_white_lists(origin, url):
                 return
 
+            # This condition seems strange. It was added when adding support for X_FORWARDED_HOST.
+            # There was already a unit test that asserted that if HTTP_HOST was missing from the
+            # request, then the referer would not be replaced. This test previously passed because
+            # in try block below we referenced request.META['HTTP_HOST'] which would throw a KeyErro
+            # and skip referer replacement. But when we changed that reference to request.get_host()
+            # in order to use X_FORWARDED_HOST when enabled, then the KeyError would never be thrown
+            # because get_host() uses the SERVER_NAME when HTTP_HOST doesn't exist. So in order to
+            # maintain that legacy functionality, the following condition was added. If moving forward,
+            # we think that we should replace referer with SERVER_NAME when HTTP_HOST is not specified,
+            # then this condition can be removed.
```

Comment:
Hi @jacobg . I saw your prompt on the issue 28 days ago, sorry I didn't get back to you, have been busy.

This massive comment and weird logic scares me, and I don't quite have the time to research into this right now. I think I'll trust you here, but please don't add a giant comment to the source, instead write the explanation in the PR, and maybe cover with extra tests.

Could you remove the comment, and maybe cover more conditions with tests if necessary? Also please add a note in HISTORY.rst saying "Use Django's request.get_host() to resolve host instead of request.META['HTTP_HOST']. This adds support for the X_FORWARDED_HOST header if configured in Django".

And if you think there's a future where the condition could be removed, please open a new issue with a similarly detailed explanation as to why it could go.